matchzoo/models/cdssm_model.py

```diff
@@ -0,0 +1,95 @@
+"""An implementation of CDSSM (CLSM) model."""
+
+from matchzoo import engine
+
+from keras import Model
+from keras.layers import Input, Conv1D, MaxPooling1D, Dot, Dense, Flatten
+
+
+class CDSSMModel(engine.BaseModel):
+    """
+    Convolutional deep structured semantic model.
+
+    Examples:
+        >>> model = CDSSMModel()
+        >>> model.guess_and_fill_missing_params()
+        >>> model.build()
+
+    """
+
+    @classmethod
+    def get_default_params(cls) -> engine.ParamTable:
+        """:return: model default parameters."""
+        params = super().get_default_params()
+        params['optimizer'] = 'sgd'
+        # TODO GET TRI-LETTER DIMENSIONALITY FROM FIT-TRANSFORM AS INPUT SHAPE
+        # Dimension: (NUM_TRI_LETTERS, Contextual Sliding Window )
+        params['input_shapes'] = [(10, 30000), (20, 30000)]
+        params.add(engine.Param('w_initializer', 'glorot_normal'))
+        params.add(engine.Param('b_initializer', 'zeros'))
+        params.add(engine.Param('dim_fan_out', 128))
+        params.add(engine.Param('dim_conv', 300))
+        params.add(engine.Param('contextual_window', 3))
+        params.add(engine.Param('strides', 1))
+        params.add(engine.Param('padding', 'same'))
+        params.add(engine.Param('activation_hidden', 'tanh'))
+        return params
+
+    def _create_base_network(self, input_shape: tuple) -> Model:
+        """
+        Apply conv and maxpooling operation towards to each tri-letter.
+
+        The input shape is `num_word_ngrams`*(`contextual_window`*
+        `dim_triletter`), as described in the paper, `contextual_window`
+        is 3, according to their observation, `dim_triletter` is about
+        30,000, so each `word_ngram` is a 1 by 90,000 matrix.
+
+        :param input_shape: tuple of input shapes.
+        :return: Keras `Model`, input 1 by n dimension tensor, output
+                 128d tensor.
+        """
+        # Input word hashing layer.
+        x_in = Input(shape=input_shape)
+        # Apply 1d convolutional on each word_ngram (lt).
+        # Input shape: (batch_size, num_tri_letters, 90000)
+        # Sequence of num_tri_letters vectors of 90000d vectors.
+        x = Conv1D(filters=self._params['dim_conv'],
+                   kernel_size=self._params['contextual_window'],
+                   strides=self._params['strides'],
+                   padding=self._params['padding'],
+                   activation=self._params['activation_hidden'],
+                   kernel_initializer=self._params['w_initializer'],
+                   bias_initializer=self._params['b_initializer'])(x_in)
+        # Apply max pooling by take max at each dimension across
+        # all word_trigram features.
+        x = MaxPooling1D(pool_size=input_shape[1],
+                         padding=self._params['padding'])(x)
+        # Apply a none-linear transformation use a tanh layer.
+        x = Flatten()(x)
+        x_out = Dense(self._params['dim_fan_out'],
```

Comment:
The number of full-connected layers should be able to customized.  At the same time, the number of hidden nodes also need to be defined by users.